nsot/models.py

```diff
@@ -1396,8 +1396,43 @@ def set_addresses(self, addresses, overwrite=False, partial=False):
 
         self.clean_addresses()
 
+    def get_ancestors(self):
+        """Recursively get all parents of an interface"""
+        p = self.parent
+        ancestors = []
+        while p is not None:
+            ancestors.append(p)
+            p = p.parent
+        return ancestors
+
+    def get_children(self):
+        """Return the immediate children of an interface"""
+        return Interface.objects.filter(parent=self)
+
+    def get_descendants(self):
+        """Recursively return all the children of an interface"""
+        s = list(self.get_children())
+        children = []
+        while len(s) > 0:
+            top = s.pop()
+            children.append(top)
+            for c in top.get_children():
+                s.append(c)
+        return children
+
+    def get_root(self):
+        """Return the parent of all ancestors of an interface"""
+        root = self
+        while root.parent is not None:
+            root = root.parent
+        return root
+
+    def get_siblings(self):
+        """Return the interfaces with same parent as an interface"""
+        return list(Interface.objects.filter(parent=self.parent, device=self.device).exclude(id=self.id))
```

Comment:
Just a comment: Last but not least, casting the queryset to a list may become problematic in the future in the event we need to implement filtering or desire to chain filters on these results.